File: _C_satellite/matlab/hw3/satelliteSim.m
satelliteParamHWB  % load parameters

% instantiate satellite, and reference input classes 
satellite = satelliteDynamics(P);  
addpath('../hw2'); reference = signalGenerator(0.5, 0.1);
addpath('../hw2'); torque = signalGenerator(.1, .1);

% instantiate the data plots and animation
addpath('../hw2'); dataPlot = dataPlotter(P);
addpath('../hw2'); animation = satelliteAnimation(P);

% main simulation loop
t = P.t_start;  % time starts at t_start
while t < P.t_end  
    % Propagate dynamics in between plot samples
    t_next_plot = t + P.t_plot;
    while t < t_next_plot % updates control and dynamics at faster simulation rate
        r = reference.square(t);
        u = torque.sin(t);  % Calculate the input force
        y = satellite.update(u);  % Propagate the dynamics
        t = t + P.Ts; % advance time by Ts
    end
    % update animation and data plots
    animation.update(satellite.state);
    dataPlot.update(t, r, satellite.state, u);
end

File: _C_satellite/python/hw3/satelliteSim.py
import matplotlib.pyplot as plt
import sys
sys.path.append('..')  # add parent directory
import satelliteParam as P
from hw2.signalGenerator import signalGenerator
from hw2.satelliteAnimation import satelliteAnimation
from hw2.dataPlotter import dataPlotter
from satelliteDynamics import satelliteDynamics

# instantiate satellite, controller, and reference classes
satellite = satelliteDynamics()
reference = signalGenerator(amplitude=0.5, frequency=0.1)
torque = signalGenerator(amplitude=0.1, frequency=0.1)

# instantiate the simulation plots and animation
dataPlot = dataPlotter()
animation = satelliteAnimation()

t = P.t_start  # time starts at t_start
while t < P.t_end:  # main simulation loop

    # Propagate dynamics in between plot samples
    t_next_plot = t + P.t_plot

    # updates control and dynamics at faster simulation rate
    while t < t_next_plot:  
        r = reference.square(t)
        u = torque.sin(t)
        y = satellite.update(u)  # Propagate the dynamics
        t = t + P.Ts  # advance time by Ts

    # update animation and data plots
    animation.update(satellite.state)
    dataPlot.update(t, r, satellite.state, u)

    # the pause causes the figure to be displayed during the
    # simulation
    plt.pause(0.0001)  

# Keeps the program from closing until the user presses a button.
print('Press key to close')
plt.waitforbuttonpress()
plt.close()
